src/_includes/components/events-banner.njk

```diff
@@ -1,15 +1,63 @@
+            <span class="hidden sm:block">-</span>
+            <span class="flex items-center gap-1 font-medium underline" >
+                {{ event.buttonText }} {% include "components/icons/chevron-right-sm.svg" %}
+            </span>
+        </a>
+        {% endif %}
+    {% endfor %}
+</div>
+
 <script>
-</script>
+    window.onload = function() {
+        const banners = document.querySelectorAll('.event-banner');
+        const container = document.querySelector('.event-banner-container');
+        let currentBanner = 0;
+
+        // Change the container color to match the current banner
+        function changeContainerColor(banner) {
+            const colorClass = banner.className.match(/bg-\w+-\d+/)[0];
+            const hoverColorClass = banner.className.match(/hover:bg-\w+-\d+/)[0];
+            container.className = 'event-banner-container ' + colorClass + ' ' + hoverColorClass + ' hidden sm:block';
+        }
+
+        // All banners are hidden by default
+        banners.forEach(banner => banner.style.visibility = 'hidden');
 
-<a data-action="event-banner" href="/webinars/2024/deploy-flowfuse-on-industrial-iot-with-ncd-io/"
-    class="hidden md:flex text-center w-full py-1 pb-3 sm:py-3 bg-indigo-700 text-indigo-100 text-sm px-2 gap-1 sm:gap-2 flex-col sm:flex-row items-center justify-center hover:bg-indigo-600">
-    <span class="hidden sm:block font-semibold">Webinar</span>
-    <span class="hidden sm:block">-</span>
-    <span class="">     
-        Deploy FlowFuse on Industrial IoT with NCD.io
-    </span>
-    <span class="hidden sm:block">-</span>
-    <span class="flex items-center gap-1 font-medium underline" >
-        Learn more {% include "components/icons/chevron-right-sm.svg" %}
-    </span>
-</a>
+        // Show the first banner
+        banners[0].style.visibility = 'visible';
+        changeContainerColor(banners[0]);
+
+        // If there is more than one banner, change the visible banner every 5 seconds
+        if (banners.length > 1) {
+            setInterval(function() {
+                // Hide the current banner
+                banners[currentBanner].style.visibility = 'hidden';
+
+                // Increment the current banner index
+                currentBanner = (currentBanner + 1) % banners.length;
+
+                // Show the new banner
+                banners[currentBanner].style.visibility = 'visible';
+                banners[currentBanner].style.animation = 'none'; // reset animation
+                // force reflow, triggering a new animation
+                void banners[currentBanner].offsetWidth;
+                banners[currentBanner].style.animation = 'slideDown 0.5s ease';
+
+                // Change the container color to match the new banner
+                setTimeout(function() {
+                    changeContainerColor(banners[currentBanner]);
+                }, 500); // wait for the slideDown animation to finish
+            }, 7000);
```

Comment:
7 seconds isn't a lot imo, do we want to push this back?

Reply:
I've changed it to 10s